### DISCUSSION TITLE
Is Custom Java Projects View possible?

### DISCUSSION THREAD

**Discussion by BarryDrez** [redhat-developer/vscode-java#3026] [Q&A]
Hi - This may be a question for the Theia community, but I was hoping maybe I can get some insights here.  Can anyone please tell me if there is a way to create a custom Theia view that contains only the JAVA PROJECTS section of the Explorer/Navigator view?  Mainly, I just want to display the JRE System Library as well as the Referenced Libraries.  I would like to create a separate view on the bottom (I know how to do this) that contains these items.

**Comment by Eskibear** [comment]
AFAIK, in latest vscode, all views are draggable, including the file explorer. You can create a view and drag whatever you want into it.

**Reply by BarryDrez** [reply]
Hi @Eskibear. Thanks for the reply.  In my case, I am hiding the file explorer, so I need a way to do this programmatically within my Theia extension.  Do you know if that is doable?

**Reply by Eskibear** [reply]
Seems there's no programmatical API for changing  layout. You may check if you can do something with [Profiles](https://code.visualstudio.com/docs/editor/profiles) to achive that.

**Reply by BarryDrez** [reply]
@Eskibear Actually, what I would like to do is create a brand new view that displays the Java Project info.  Can you point me to the code that drills down into the Jar files and creates the tree view that shows the packages and classes?

**Reply by Eskibear** [reply]
Java Projects view is provided by this extension, https://github.com/microsoft/vscode-java-dependency/ I would suggest you open an issue describing your use case there. See if they can provide you the flexibility if the use case is valid.

**Reply by BarryDrez** [reply]
Thank you

**Reply by BarryDrez** [reply]
Thanks
